cfme/fixtures/base.py

```diff
@@ -40,17 +40,21 @@ def fix_merkyl_workaround(request, appliance):
 def fix_missing_hostname(appliance):
     """Fix for hostname missing from the /etc/hosts file
 
-    Note: Affects RHOS-based appliances but can't hurt the others so
-          it's applied on all.
+    Note: Needed primarily on RHOS based providers. Negatively impacts freeipa (and probably
+    other services) by setting a FQDN hostname to the loopback IPv4 address.  RHOS providers do
+    not have a FQDN returned by `hostname`
+
+    IPAppliance 'hostname' attribute
     """
     if isinstance(appliance, DummyAppliance) or appliance.is_dev:
         return
     ssh_client = appliance.ssh_client
     logger.info("Checking appliance's /etc/hosts for its own hostname")
     if ssh_client.run_command('grep $(hostname) /etc/hosts').rc != 0:
         logger.info("Adding it's hostname to its /etc/hosts")
-        # Append hostname to the first line (127.0.0.1)
-        ret = ssh_client.run_command('sed -i "1 s/$/ $(hostname)/" /etc/hosts')
+        # Append hostname to the /etc/hosts
+        ret = ssh_client.run_command('echo "{ip}  $(hostname)" >> /etc/hosts'
```

Comment:
**QUESTION:** There is the possibility that "$some_ip $(hostname)" is already in the /etc/hosts file.    In that case wouldn't you want to edit the line with the wrong IP (or remove it and then do this), or does it only mater that there is a line with the appliance IP and it's hostname?